src/ts/components/dashboard/TopMenu.tsx

```diff
@@ -1,33 +1,47 @@
 import * as React from 'react'
 import { connect } from 'react-redux'
 import { withRouter, RouteComponentProps } from 'react-router'
-import { Image, Grid } from 'semantic-ui-react'
+import { Image, Grid, Dimmer, Loader } from 'semantic-ui-react'
 import { networks } from '../../constants/networks'
+import { WsProvider } from '@polkadot/rpc-provider'
 import { IAppState } from '../../background/store/all'
 import { saveSettings } from '../../background/store/settings'
 import { getTransactions } from '../../background/store/transaction'
-import { ChainDropdown } from '../basic-components'
+import {ChainDropdown} from '../basic-components'
 import styled from 'styled-components'
 import SettingsMenu from './SettingsMenu'
+import { ApiOptions } from "@polkadot/api/types";
+import { createApi, destroyApi } from "../../background/store/api-context";
 
 interface ITopMenuProps extends StateProps, DispatchProps, RouteComponentProps {}
 
 interface ITopMenuState {
+  isLoading: boolean,
```

Comment:
Can we remove this flag and instead use apiContext.apiReady as Dimer's toggling flag? This way user does not have to wait at least 3 seconds?

Reply:
it still need 3 sec regardless. but we probably can save a flag.

Reply:
The `isLoading` can be set to `false` in at least 3 seconds when the timeout happens, while `apiContext.apiReady` is set to `true` by `redux` as soon as the api is created, correct me if I am wrong.

Reply:
I see. you are right.

Reply:
updated